### ISSUE TITLE
(ITA)How to concat train dataset and whether the visual context is also adopted on the dev dataset or test dataset

### ISSUE THREAD

**Issue by 257556227** [Alibaba-NLP/KB-NER#15]
Thank you for your great work!
I have read your code a little, but it is difficult to find the core code related to ITA.  Thus, there is still a lot of confusion after browsing the yaml configuration and **finetune_train.py**.
1. I took a look at the data set and it looks like it's all from the `Corpus: ColumnCorpus-SNAP:ColumnCorpus-SNAPDOC`. The former is the common NER dataset, and the latter is the concatenation of the NER dataset with ocr, region, and global text. **But I want to know how to cancat the train dataset?** [(url:https://github.com/Alibaba-NLP/KB-NER/blob/main/flair/trainers/finetune_trainer.py#:~:text=train_data%20%3D%20%5Bself.corpus.train%5D)](https://github.com/Alibaba-NLP/KB-NER/blob/4d8557abc637d35b7e25dcf7b26177432475fae0/flair/trainers/finetune_trainer.py#L586) Just through the `ConcatDataset`? Also, do the first Corpus and the next corpus go into training together?
2.  doc_caption_vinvl_classattr_ocr dataset consists of ocr, 5 global captions and region labels. Will them concatenate with raw sentence beyond **_max_seq_len_**?  If it's possible, is there a truncation operation, and **eos**_token delimited clauses should have global attention between them？
3.  Although _train with dev_ not in ITA yaml, i also want to know the format of dev_dataloader and test_dataloader. Only raw ner dataset?
4.  If possible, could you give me same training(dev/test) data format case which feed into xlm-robert-large. I would be greatly grateful.

**Comment by wangxinyu0922** [comment]
1. You may read these lines for more information https://github.com/Alibaba-NLP/KB-NER/blob/4d8557abc637d35b7e25dcf7b26177432475fae0/flair/config_parser.py#L298-L344 . The two corpora are trained together.
2. Yes. Taking 5 captions as an example, if I want to add the second sentence into the new sentence but the subtoken length exceeds 510, this sentence will be dropped and move to the third sentence (and repeat). The eos token will be added to the final sequence during the encoding of transformer models.
3. The formatting of the dev and test is the same as the training set.
4. I'm not sure of your meaning. The datasets are released here https://github.com/Alibaba-NLP/KB-NER/tree/main/ITA#datasets

**Comment by 257556227** [comment]
Thank you very much for your answer,
Wish you a happy life！

**Comment by schicksa** [comment]
@wangxinyu0922 @257556227  
I also have a question about eos token, which often used in generative frameworks such as GPT, and it passes through the Tokenizer.convert_ tokens_ to_ ids of bert or roberta is [UNK] or<unk>. 
Can it still serve as a truncation symbol?

**Comment by 257556227** [comment]
@schicksa
我感觉应该不影响吧，[UNK]就是未在词表出现的词，<eos>本身也无实义的，而且我感觉这些图转文信息主要还是减少图文gap，作为辅助信息来增强文本表示的，实际上在过transformer的时候，attention_mask和input_id序列会增长，同时激活值也会成正比增加，评估的时候应该还是只看纯T部分。不过这样的话，dataloader加载的时候，sequence length大概会增加5-8倍，显存很容易溢出，max bsz得减少到1/4左右来控制，算是trade off吧。

**Comment by schicksa** [comment]
你可能没懂我的意思，我的意思是EOS token有什么作用，如果用Roberta的分词器的话，segment_id应该都是0，那么，采用EOS token分开的意义不就只是个象征作用，无实际意义吗
代码中EOS token暂时只找到这两处
	```
if '<EOS>' in words:
		eos_id = words.index('<EOS>')
		current_sentence.chunk_sentence(0,eos_id)
```
`tokens = tokenizer.tokenize(s + "<eos>")`
前者是将第一个<EOS>token前的句子取出来，也就是UMT中的data
后者就是测试用，一个句子加<eos>进行tokenize的结果
所以，我认为应该只是让模型在训练的时候学到它们是分隔符来着。
**对了，还有个问题想问下，Cross-view-alignment是怎么实现的啊，forward-backword-score这部分嵌套有点多，没看太懂**

**Comment by wangxinyu0922** [comment]
> 你可能没懂我的意思，我的意思是EOS token有什么作用，如果用Roberta的分词器的话，segment_id应该都是0，那么，采用EOS token分开的意义不就只是个象征作用，无实际意义吗 代码中EOS token暂时只找到这两处
> 
> ```
> if '<EOS>' in words:
>       eos_id = words.index('<EOS>')
>       current_sentence.chunk_sentence(0,eos_id) #1
> 
> tokens = tokenizer.tokenize(s + "<eos>") #2
> ```
> 
> 前者是将第一个token前的句子取出来，也就是UMT中的data 后者就是测试用，一个句子加进行tokenize的结果 所以，我认为应该只是让模型在训练的时候学到它们是分隔符来着。 **对了，还有个问题想问下，Cross-view-alignment是怎么实现的啊，forward-backword-score这部分嵌套有点多，没看太懂**

在代码里面，<EOS> token在tokenize之前会转换成对应transformer 模型的sep token（BERT里就是[SEP]，roberta就是</s>）。主要目的还是一个分割提示符的作用。cross-view-alignment就是把在CRF上计算每个位置的后验概率分布（forward-backward-score然后normalize），最后计算KL divergence。

**Comment by schicksa** [comment]
> > 你可能没懂我的意思，我的意思是EOS token有什么作用，如果用Roberta的分词器的话，segment_id应该都是0，那么，采用EOS token分开的意义不就只是个象征作用，无实际意义吗 代码中EOS token暂时只找到这两处
> > ```
> > if '<EOS>' in words:
> >       eos_id = words.index('<EOS>')
> >       current_sentence.chunk_sentence(0,eos_id) #1
> > 
> > tokens = tokenizer.tokenize(s + "<eos>") #2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 前者是将第一个token前的句子取出来，也就是UMT中的data 后者就是测试用，一个句子加进行tokenize的结果 所以，我认为应该只是让模型在训练的时候学到它们是分隔符来着。 **对了，还有个问题想问下，Cross-view-alignment是怎么实现的啊，forward-backword-score这部分嵌套有点多，没看太懂**
> 
> 在代码里面， token在tokenize之前会转换成对应transformer 模型的sep token（BERT里就是[SEP]，roberta就是）。主要目的还是一个分割提示符的作用。cross-view-alignment就是把在CRF上计算每个位置的后验概率分布（forward-backward-score然后normalize），最后计算KL divergence。

感谢解答
